src/main/scala/mesosphere/marathon/core/launchqueue/LaunchStats.scala

```diff
@@ -170,7 +170,7 @@ object LaunchStats extends StrictLogging {
   ) {
     def queueDelay(clock: Clock): QueueDelay = {
       val timeLeft = this.backOffUntil.map(clock.now() until _).getOrElse(0.seconds)
-      val overdue = timeLeft.toSeconds < 0
+      val overdue = backOffUntil.isEmpty || timeLeft.toSeconds < 0
```

Comment:
🤔 so if my app is backed off, `backOffUntil` is not empty, right? But if we're past the delay and we should be launching, this will still be not empty but we're still overdue now, right?

I don't know how are people using our API because even I struggle with understanding what the field mean.

Reply:
backOffUntil is not exposed via the API. For the user of the api, there's 2 fields. See the updated docs above:
`timeLeft`: The number of seconds to wait, before the next launch attempt is started.
`overdue`: true, if this runspec is waiting on offers. If false, it means runspec is backed off and waits until timeLeftSeconds for the next launch attempt.

In other words, task launches are overdue when there is no backoff delay, or when the point until we were delayed is in the past.

I don't think this was ever meant to be _used_, rather to help debugging. I agree it's not even close to being intuitive :D

Reply:
+1 to above ^.

> so if my app is backed off, backOffUntil is not empty, right?
yes.

> But if we're past the delay and we should be launching, this will still be not empty but we're still overdue now, right?
If the delay is not set, the option becomes empty, overdue becomes true and we will be launching. If the option is defined then we check if the timeLeft is < 0 and then we will be launching (or not) based on the value of timeLeft. 

as @meichstedt  mentioned, this is not exposed to end user. only `overdue` and `timeLeft` are exposed.

Reply:
I had sth similar on the first draft of this PR and although it entails identical behavior - after discussing with @timcharper, felt this is more descriptive.

Reply:
@takirala I see, I guess it's matter of taste. This was VERY CONFUSING to me upon reviewing because to be sure this works I had to figure out when is backOffUntil None and when it has a value and what it means...